alarm_trap_sender.cpp

```diff
@@ -50,37 +50,28 @@ AlarmFilter AlarmFilter::_instance;
 
 AlarmTrapSender AlarmTrapSender::_instance;
 
-bool ActiveAlarmList::update(AlarmTableDef& alarm_table_def, const std::string& issuer)
+bool ObservedAlarms::update(AlarmTableDef& alarm_table_def, const std::string& issuer)
 {
   bool updated = false;
 
   std::map<unsigned int, AlarmListEntry>::iterator it = _index_to_entry.find(alarm_table_def.alarm_index());
-
-  if (alarm_table_def.is_not_clear())
-  {
-    if ((it == _index_to_entry.end()) || (alarm_table_def.severity() != it->second.alarm_table_def().severity()))
-    {
-      _index_to_entry[alarm_table_def.alarm_index()] = AlarmListEntry(alarm_table_def, issuer);
-      updated = true;
-    }
-  }
-  else
+  
+  // If either the current alarm doesn't exist in the ObservedAlarms mapping
+  // or there is an entry for the current alarm in the mapping but at a
+  // different severity to the one we are currently raising the alarm with, we
+  // want to update the mapping and send a trap for the alarm.
```

Comment:
Please could you remove the reference to sending a trap - as I said before, this class doesn't know about traps.